frontend/src/components/NavDropdown.tsx

```diff
@@ -0,0 +1,57 @@
+"use client";
+
+import clsx from "clsx";
+import { IndexType } from "src/types/generalTypes";
+
+import { Dispatch, JSX, SetStateAction, useEffect, useState } from "react";
+import { Menu, NavDropDownButton } from "@trussworks/react-uswds";
+
+interface NavDropdownProps {
+  activeNavDropdownIndex: IndexType;
+  index: number;
+  isCurrent: boolean;
+  linkText: string;
+  menuItems: JSX.Element[];
+  setActiveNavDropdownIndex: Dispatch<SetStateAction<IndexType>>;
+}
+
+export default function NavDropdown({
+  activeNavDropdownIndex,
+  index,
+  isCurrent,
+  linkText,
+  menuItems,
+  setActiveNavDropdownIndex,
+}: NavDropdownProps): JSX.Element {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  function handleToggle() {
+    const activeIndex: IndexType = isOpen ? null : index;
```

Comment:
I think we can manage this without the local state. 
```suggestion
    const newIndex = activeNavDropdownIndex === index ? null : index;
    setActiveNavDropdownIndex(newIndex);
```

would also need to update line 52 accordingly